### ISSUE TITLE
change cifar10 to dagnn,the objective is NaN

### ISSUE THREAD

**Issue by anewlearner** [vlfeat/matconvnet#550]
Dagnn can  be change from simplenn by the function dagnn.DagNN.fromSimpleNN.
I don't want to use this method,but get dagnn directly. My code is as follows. But the objective is NaN. Could someone can help me to fix this? Thanks.
`function net = cnn_cifar_initdag(varargin)
%cifar10
net = dagnn.DagNN();
% net.meta.inputSize = [32 32 3] ;
net.meta.trainOpts.learningRate = [0.05*ones(1,30) 0.005*ones(1,10) 0.0005*ones(1,5)] ;
net.meta.trainOpts.weightDecay = 0.0001 ;
net.meta.trainOpts.batchSize = 100 ;
net.meta.trainOpts.numEpochs = numel(net.meta.trainOpts.learningRate) ;

%block1
net.addLayer('conv1', dagnn.Conv('size', [5 5 3 32], 'hasBias', true, 'stride', 1, 'pad', 2), {'input'}, {'conv1'},  {'conv1f'  'conv1b'});
net.addLayer('pool1', dagnn.Pooling('method', 'max', 'poolSize', [3,3], 'stride', 2, 'pad', [0 1 0 1]), {'conv1'}, {'pool1'}, {});
net.addLayer('relu1', dagnn.ReLU(), {'pool1'}, {'relu1'}, {});
%block2
net.addLayer('conv2', dagnn.Conv('size', [5 5 32 32], 'hasBias', true, 'stride', 1, 'pad', 2), {'relu1'}, {'conv2'},  {'conv2f'  'conv2b'});
net.addLayer('relu2', dagnn.ReLU(), {'conv2'}, {'relu2'}, {});
net.addLayer('pool2', dagnn.Pooling('method', 'avg', 'poolSize', [3,3], 'stride', 2, 'pad', [0 1 0 1]), {'relu2'}, {'pool2'}, {});
%block3
net.addLayer('conv3', dagnn.Conv('size', [5 5 32 64], 'hasBias', true, 'stride', 1, 'pad', 2), {'pool2'}, {'conv3'},  {'conv3f'  'conv3b'});
net.addLayer('relu3', dagnn.ReLU(), {'conv3'}, {'relu3'}, {});
net.addLayer('pool3', dagnn.Pooling('method', 'avg', 'poolSize', [3,3], 'stride', 2, 'pad', [0 1 0 1]), {'relu3'}, {'pool3'}, {});
%block4
net.addLayer('conv4', dagnn.Conv('size', [4 4 64 64], 'hasBias', true, 'stride', 1, 'pad', 0), {'pool3'}, {'conv4'},  {'conv4f'  'conv4b'});
net.addLayer('relu4', dagnn.ReLU(), {'conv4'}, {'relu4'}, {});

%block5
net.addLayer('prediction', dagnn.Conv('size', [1 1 64 10], 'hasBias', true, 'stride', 1, 'pad', 0), {'relu4'}, {'prediction'},  {'conv5f'  'conv5b'});

net.addLayer('objective', dagnn.Loss('loss', 'softmaxlog'), {'prediction','label'}, {'objective'}, {});
net.addLayer('error', dagnn.Loss('loss', 'classerror'),{'prediction','label'}, 'error') ;
net.initParams();
`
